lms/djangoapps/django_comment_client/utils.py

```diff
@@ -741,35 +752,42 @@ def get_group_id_for_comments_service(request, course_key, commentable_id=None):
     Raises:
         ValueError if the requested group_id is invalid
     """
-    if commentable_id is None or is_commentable_divided(course_key, commentable_id):
+    course_discussion_settings = get_course_discussion_settings(course_key)
+    if commentable_id is None or is_commentable_divided(course_key, commentable_id, course_discussion_settings):
         if request.method == "GET":
             requested_group_id = request.GET.get('group_id')
         elif request.method == "POST":
             requested_group_id = request.POST.get('group_id')
         if has_permission(request.user, "see_all_cohorts", course_key):
             if not requested_group_id:
                 return None
-            try:
-                group_id = int(requested_group_id)
-                get_cohort_by_id(course_key, group_id)
-            except CourseUserGroup.DoesNotExist:
-                raise ValueError
+            group_id = int(requested_group_id)
+            _verify_group_exists(group_id, course_discussion_settings)
         else:
             # regular users always query with their own id.
-            group_id = get_group_id_for_user(request.user, course_key)
+            group_id = get_group_id_for_user(request.user, course_discussion_settings)
         return group_id
     else:
-        # Never pass a group_id to the comments service for a non-cohorted
+        # Never pass a group_id to the comments service for a non-divided
         # commentable
         return None
 
 
-def get_group_id_for_user(user, course_key):
+def get_group_id_for_user(user, course_discussion_settings):
     """
-    This method will be modified to consider Enrollment Tracks.
-    Currently pass-through to get_cohort_id.
+    Given a user, return the group_id for that user according to the course_discussion_settings.
+    If discussions are not divided, this method will return None.
+    It will also return None if the user is in no group within the specified division_scheme.
     """
-    return get_cohort_id(user, course_key)
+    division_scheme = _get_course_division_scheme(course_discussion_settings)
+    if division_scheme == DIVIDED_BY_COHORT:
+        return get_cohort_id(user, course_discussion_settings.course_id)
+    elif division_scheme == DIVIDED_BY_ENROLLMENT_TRACK:
+        partition_service = PartitionService(course_discussion_settings.course_id)
+        group_id = partition_service.get_user_group_id_for_partition(user, ENROLLMENT_TRACK_PARTITION_ID)
+        return -1 * group_id if group_id is not None else None
```

Comment:
`if group_id is not None else None`

can be simplified to 

`if group_id else None`